File: chat-model-experiments/openai-features/chat_with_docs.js
import fs from "fs";
import OpenAI from "openai";
import "dotenv/config";
const openai = new OpenAI({
  apiKey: process.env.OPEN_AI_API_KEY,
});

const file = await client.files.create({
  file: fs.createReadStream("draconomicon.pdf"),
  purpose: "user_data",
});

const completion = await client.chat.completions.create({
  model: "gpt-4.1",
  messages: [
    {
      role: "user",
      content: [
        {
          type: "file",
          file: {
            file_id: file.id,
          },
        },
        {
          type: "text",
          text: "What is the first dragon in the book?",
        },
      ],
    },
  ],
});

console.log(completion.choices[0].message.content);

File: chat-model-experiments/backend/services/chat_with_docs.py
from openai import OpenAI
import os

# MCQ Model

API_KEY = os.environ.get("OPENAI_API_KEY")

client = OpenAI(
    api_key=API_KEY
)


def talk_to_llm(input_text, system_prompt, file_id=None, previous_response_id=None, response_format=None):
    
    if response_format:
        response = client.responses.parse(
            model="gpt-4.1",
            previous_response_id=previous_response_id,
            input=[
                {
                    "role": "system",
                    "content": [
                        {
                            "type": "input_text",
                            "text": system_prompt
                        }
                    ]
                },
                {
                    "role": "user",
                    "content": [
                        {
                            "type": "input_file",
                            "file_id": file_id,
                        },
                        {
                            "type": "input_text",
                            "text": input_text,
                        },
                    ]
                }
            ],
            text_format=response_format,
        )
    
    else:
        response = client.responses.create(
            model="gpt-4.1",
            previous_response_id=previous_response_id,
            input=[
                {
                    "role": "system",
                    "content": [
                        {
                            "type": "input_text",
                            "text": system_prompt
                        }
                    ]
                },
                {
                    "role": "user",
                    "content": [
                        {
                            "type": "input_file",
                            "file_id": file_id,
                        },
                        {
                            "type": "input_text",
                            "text": input_text,
                        },
                    ]
                }
            ],
        )
        
    return response
